### PR TITLE
fix(gate): make Copilot presence detection review-based, not assignee-based

### LINKED ISSUE #1670: fix(gate): Copilot-presence detection must be review-based, not assignee-based (false 'Copilot not enabled' on reviewer-configured repos)
## Context

Final rc.5 retrospective. The #1613 subagent misdiagnosed "Copilot code review is not enabled on this repo" — but Copilot (`copilot-pull-request-reviewer`) actually reviewed all 14 drive PRs (1-2 reviews each). The misdiagnosis used an assignee-based proxy; Copilot is a REVIEWER on this repo, never an assignee, so a null assignee-match was misread as "Copilot absent."

## Problem

Copilot-presence detection that keys off assignee membership produces false-negatives when Copilot is configured as a reviewer (not assignee). This caused a wrong "Copilot not enabled" claim and could cause the gate to skip the Copilot-convergence requirement under a false premise.

## Proposal

Make Copilot-presence detection review-based: check the PR's requested reviewers / submitted reviews for `copilot-pull-request-reviewer[bot]`, not the assignee list. Align the "is Copilot enabled / did Copilot review" check with the actual review surface.

## Acceptance criteria
- [x] Copilot-presence detection is review-based (requested reviewers + submitted reviews), not assignee-based.
- [x] A repo where Copilot is a reviewer (not assignee) correctly reports Copilot as present.
- [x] The gate's Copilot-convergence requirement is not skipped under a false "Copilot absent" premise.
- [x] Regression test covering the reviewer-not-assignee case.
- [x] `npm run verify` green.

## Non-goals
- Changing how Copilot is requested or configured.
- Changing the gate's Copilot-convergence contract.

### PR DESCRIPTION
**Pull request by mfittko**
## Summary

Closes #1670. Makes Copilot-presence detection **review-based** (requested reviewers + submitted reviews), never assignee-based.

## Context

The final rc.5 retrospective flagged a recurring misdiagnosis: a subagent claimed "Copilot code review is not enabled on this repo" — but Copilot (`copilot-pull-request-reviewer[bot]`) actually reviewed all 14 drive PRs. On this repo Copilot is configured as a **reviewer**, never an assignee, so an assignee-based proxy returns a null match and is misread as "Copilot absent". That false premise risks the gate skipping the Copilot-convergence requirement.

## Change

- Add a pure review-surface presence resolver `resolveCopilotReviewPresence` in `packages/core/src/github/copilot-helpers.mjs`. Presence is derived from `requested_reviewers` membership OR any submitted Copilot review on the PR — assignment is a disjoint surface and is never consulted.
- Wire it into `scripts/github/request-copilot-review.mjs`: the post-request verification now treats review-surface presence (an existing submitted Copilot review, or a requested-reviewer entry) as observable in-progress, so a reviewer-configured repo where `@copilot` is a silent no-op is **not** falsely reported as "Copilot absent / not enabled". This keeps the Copilot-convergence requirement from being skipped on a false premise.

## Acceptance criteria

- [x] Copilot-presence detection is review-based (requested reviewers + submitted reviews), not assignee-based.
- [x] A repo where Copilot is a reviewer (not assignee) correctly reports Copilot as present.
- [x] The gate's Copilot-convergence requirement is not skipped under a false "Copilot absent" premise.
- [x] Regression test covering the reviewer-not-assignee case.
- [x] `npm run verify` green.

## Definition of done

- [x] Presence detection resolved against the review surface (requested reviewers + submitted Copilot reviews), never the assignee list.
- [x] The actually-changed wiring path (`performCopilotReviewRequest` review-surface presence branch) is covered by an end-to-end regression test pinning the #1670 reviewer-configured-repo scenario.
- [x] `resolveCopilotReviewPresence` edge cases covered (dual source; malformed review entries).
- [x] Full `npm run verify` green.

## Validation command

Run `npm run verify` locally or rely on the `verify` CI suite. The reviewer-not-assignee regression is exercised by `packages/core/test/copilot-helpers.test.mjs` and the wiring-level #1670 scenario by `test/github/request-copilot-review.test.mjs` (`npm run test:core` and `npm run test:scripts`).

## Non-goals

- No change to how Copilot is requested or configured.
- No change to the gate's Copilot-convergence contract.

### CODE REVIEW

**Review by mfittko — COMMENTED** [review]
### Gate review: `draft_gate`


**Reviewed head SHA:** `deb5837b21440cc757c2ae67430363584125d7d7`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 8 angles reviewed; 1 finding (see per-angle breakdown below).

- `contract-surface` → `findings_present` (1 finding)
- `correctness` → `clean`
- `coverage` → `clean`
- `determinism` → `clean`
- `gate-evidence` → `clean`
- `input-validation` → `clean`
- `pr-description` → `clean`
- `scope` → `clean`

**Body-filed findings** (no in-diff location):

> None — every finding this round is an inline comment on this review.

**Next action:** ready-for-review

File: packages/core/src/github/copilot-helpers.mjs [hunk 1]
```diff
@@ -80,6 +80,37 @@ export function isCopilotLogin(login) {
...
+ */
+export function resolveCopilotReviewPresence({ requested = false, reviews = [] } = {}) {
+  const list = Array.isArray(reviews) ? reviews : [];
+  const sources = [];
+  if (requested === true) {
+    sources.push("requested_reviewer");
+  }
+  if (list.some((review) => isCopilotLogin(review?.author?.login))) {
+    sources.push("submitted_review");
+  }
```
[the remark this answers is not in the document]
Comment (mfittko) [reply packages/core/src/github/copilot-helpers.mjs:110]: Disposition: deferred (low severity, non-blocking). Draft_gate blockCleanOnFindingSeverities is [high], so this low finding never blocks the clean verdict. The finding concerns the `requested` vs `already-requested` status label in a silent-no-op scenario where Copilot presence comes solely from a prior submitted review. It is a cosmetic status-semantics nit: downstream routing treats both statuses identically (wait seam), so no behavioral divergence follows. Recorded in the draft_gate disposition ledger as deferred.

**Review by mfittko — COMMENTED** [review]
### Gate review: `pre_approval_gate`


**Reviewed head SHA:** `deb5837b21440cc757c2ae67430363584125d7d7`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 5 angles reviewed; no findings (see per-angle breakdown below).

- `acceptance-criteria` → `clean`
- `contradiction-lens` → `clean`
- `pr-checklist-matrix` → `clean`
- `renderer-security` → `clean`
- `yagni` → `clean`

**Body-filed findings** (no in-diff location):

> None — every finding this round is an inline comment on this review.

**Next action:** pre_approval_gate CLEAN(5 angles/3 distinct reviewers/0 findings); authorized MERGE on clean gates
